kbs/src/plugins/implementations/splitapi/manager.rs

```diff
@@ -0,0 +1,148 @@
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct Credentials {
+    pub ca_crt: Vec<u8>,
+    pub client_key: Vec<u8>,
+    pub client_crt: Vec<u8>,
+    pub server_key: Vec<u8>,
+    pub server_crt: Vec<u8>,
+}
+
+/// Credentials necessary for SplitAPI proxy server
+#[derive(Debug, serde::Serialize)]
+pub struct ServerCredentials {
+    pub key: Vec<u8>,
+    pub crt: Vec<u8>,
+    pub ca_crt: Vec<u8>,
+}
+
+/// Manages the credentials generation, handling requests
+/// from backend, and credentials persistence storage
+pub struct CertManager {
+    pub plugin_dir: PathBuf,
+    pub certificate_details: CertificateDetails,
+    pub credblob_file: PathBuf, //String,
+    pub state: Arc<RwLock<HashMap<String, Credentials>>>,
+    credential_loaded_from_file: AtomicBool,
+}
+
+impl CertManager {
+    pub fn new(
+        plugin_dir: PathBuf,
+        blob_file: String,
+        cert_details: &CertificateDetails,
+    ) -> anyhow::Result<Self> {
+        if !plugin_dir.exists() {
+            fs::create_dir_all(&plugin_dir)?;
+            log::info!("plugin dir created = {}", plugin_dir.display());
+        }
+
+        let cblob_file = plugin_dir.as_path().join(blob_file);
+
+        // Initialize the credential manager
+        Ok(Self {
+            plugin_dir,
+            certificate_details: cert_details.clone(),
+            credblob_file: cblob_file,
+            state: Arc::new(RwLock::new(HashMap::new())),
+            credential_loaded_from_file: AtomicBool::new(false),
+        })
+    }
+
+    async fn load_credentials(&self, key: &str) -> Option<Credentials> {
+        // Check if the credential is not loaded. If not, load them
+        if !self.credential_loaded_from_file.load(Ordering::SeqCst) {
+            if let Err(e) = self.load_from_file(&self.credblob_file).await {
+                log::warn!("Failed to load credentials from file: {}", e);
+                return None;
+            }
+
+            // Update the flag, this is a one-time load until kbs restarts
+            self.credential_loaded_from_file
+                .store(true, Ordering::SeqCst);
+        }
+
+        // Return the item from hashmap
+        let state = self.state.read().await;
+        state.get(key).cloned()
+    }
+
+    async fn load_from_file(&self, path: &PathBuf) -> Result<()> {
+        let data = tokio::fs::read_to_string(&path).await?;
+        let deserialized: HashMap<String, Credentials> = serde_json::from_str(&data)?;
+        let mut state = self.state.write().await;
+        *state = deserialized;
+        Ok(())
+    }
+
+    async fn save_to_file(&self, path: &PathBuf) -> Result<()> {
+        let state = self.state.read().await;
+        let serialized = serde_json::to_string(&*state)?;
+        tokio::fs::write(path, serialized).await?;
+        Ok(())
+    }
+}
+
+#[async_trait::async_trait]
+impl SplitAPIBackend for CertManager {
+    async fn get_server_credential(&self, params: &SandboxParams) -> Result<Vec<u8>> {
```

Comment:
What is the `server` here? These credentials are for the client, right? Maybe you're referring to some other server, like the SplitAPI server in the guest, but this is pretty confusing.